### PR TITLE
Added vps

### PR DESCRIPTION
**Pull request by zhanggbj**
Hi dr.max  @maximilien, 

These are all latest code for vps and it is a little big. Thanks for your review. 

/cc @mattcui  @jianqiu

### CODE REVIEW

**Review by maximilien — APPROVED** [review]
We need to avoid such massive PR. I will make this one exception but no more moving forward. I mean it.

File: src/github.com/cloudfoundry-community/vps/cmd/soft-layer-vm-pool-server/main.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"github.com/cloudfoundry-community/vps/restapi"
+)
+
+// This file was generated by the swagger tool.
+// Make sure not to overwrite this file after you generated it because all your edits would be lost!
+
+func main() {
+	swaggerSpec, err := loads.Analyzed(restapi.SwaggerJSON, "")
+	if err != nil {
+		log.Fatalln(err)
```
Comment (maximilien) [inline src/github.com/cloudfoundry-community/vps/cmd/soft-layer-vm-pool-server/main.go]: Let's add a message here to the log before fatally exiting.

File: src/github.com/cloudfoundry-community/vps/controllers/virtual_guest_controller.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+func (h *VirtualGuestController) OrderVirtualGuest(logger lager.Logger, vmFilter *models.VMFilter) (*models.VM, error){
+	logger = logger.Session("order-vm")
+	return h.db.OrderVirtualGuestToProvision(logger, *vmFilter)
+}
+
+func (h *VirtualGuestController) VirtualGuestsByDeployments(logger lager.Logger, names []string) ([]*models.VM, error) {
+	logger = logger.Session("vms-by-deployments")
+
```
Comment (maximilien) [inline src/github.com/cloudfoundry-community/vps/controllers/virtual_guest_controller.go]: Let's be consistent, new line or no new line. I'd say the latter.

**Comment by maximilien** [comment]
Hi, @zhanggbj and @jianqiu,

I took a pass but I need you guys to address a few things before I continue and eventually merge:
1. separate generated code using Swagger with non-generated code
2. separate any code you are reusing
3. divide the code into smaller PRs. Perhaps the tests can be one PR, generated code one PR, etc. you can send the PR in the order you want me to merge things.
4. address the comments I left in what I reviewed.

Please do that soon and I will get to it and finalize the review.

Best,

max

**Comment by maximilien** [comment]
@zhanggbj Travis CI is failing. Once fixed I'll review and merge. Thx

**Comment by zhanggbj** [comment]
Will submit separated PRs. So close it.

**Pull request closed without merging**